### PR TITLE
feat(NumberInput): optionally hide the label, resolves #1509

### LINKED ISSUE #1509: NumberInput: add hideLabel prop
### Summary
Add a hideLabel prop to the NumberInput component. This prop is provided on many other components and makes the component more consistent with the rest of the repo.

### Justification
Need this functionality.

### Available extra resources
I will submit a PR for this.

### PR DESCRIPTION
**Pull request by McBrosa**
Closes IBM/carbon-components-react#1509

Add hideLabel prop to NumberInput component.

#### Changelog

**NumberInput**
* add hideLabel prop

### CODE REVIEW

File: src/components/NumberInput/NumberInput.js [hunk 1]
```diff
@@ -216,3 +181,49 @@ export default class NumberInput extends Component {
     );
   }
 }
+
+NumberInput.defaultProps = {
```
Comment (asudoh) [inline src/components/NumberInput/NumberInput.js]: Any reason why you moved `defaultProps`/`propTypes`?

Comment (McBrosa) [reply src/components/NumberInput/NumberInput.js]: Reverted that change, for some reason I thought that was how the reset of the components were setup.

**Comment by joshblack** [comment]
@asudoh @tw15egan which prop convention are we following for boolean props again? Should it be `shouldHideLabel`?

**Comment by asudoh** [comment]
> which prop convention are we following for boolean props again? Should it be shouldHideLabel?

Thank you for asking @joshblack! I don't think we have a particular naming rule for boolean props in general, I see other components use `hideLabel` for similar purposes.

**Pull request merged**